expression/builtin_time_test.go

```diff
@@ -1255,12 +1255,15 @@ func (s *testEvaluatorSuite) TestStrToDate(c *C) {
 		Date    string
 		Format  string
 		Success bool
-		Expect  time.Time
+		Expect  types.Time
 	}{
-		{"20161122165022", `%Y%m%d%H%i%s`, true, time.Date(2016, 11, 22, 16, 50, 22, 0, time.Local)},
-		{"2016 11 22 16 50 22", `%Y%m%d%H%i%s`, true, time.Date(2016, 11, 22, 16, 50, 22, 0, time.Local)},
-		{"16-50-22 2016 11 22", `%H-%i-%s%Y%m%d`, true, time.Date(2016, 11, 22, 16, 50, 22, 0, time.Local)},
-		{"16-50 2016 11 22", `%H-%i-%s%Y%m%d`, false, time.Time{}},
+		{"20161122165022", `%Y%m%d%H%i%s`, true, types.Time{Time: types.FromDate(2016, 11, 22, 16, 50, 22, 0), Type: mysql.TypeDatetime, Fsp: 0}},
+		{"2016 11 22 16 50 22", `%Y%m%d%H%i%s`, true, types.Time{Time: types.FromDate(2016, 11, 22, 16, 50, 22, 0), Type: mysql.TypeDatetime, Fsp: 0}},
+		{"16-50-22 2016 11 22", `%H-%i-%s%Y%m%d`, true, types.Time{Time: types.FromDate(2016, 11, 22, 16, 50, 22, 0), Type: mysql.TypeDatetime, Fsp: 0}},
+		{"16-50 2016 11 22", `%H-%i-%s%Y%m%d`, false, types.Time{Time: types.FromDate(2016, 11, 22, 16, 50, 22, 0), Type: mysql.TypeDatetime, Fsp: 0}},
+		{"16-50-22-01 2016 11 22", `%H-%i-%s-%f%Y%m%d`, true, types.Time{Time: types.FromDate(2016, 11, 22, 16, 50, 22, 1), Type: mysql.TypeDatetime, Fsp: 6}},
+		{"16-50-22-123 2016 11 22", `%H-%i-%s-%f%Y%m%d`, true, types.Time{Time: types.FromDate(2016, 11, 22, 16, 50, 22, 123), Type: mysql.TypeDatetime, Fsp: 6}},
+		{"16-50-22-123456 2016 11 22", `%H-%i-%s-%f%Y%m%d`, true, types.Time{Time: types.FromDate(2016, 11, 22, 16, 50, 22, 123456), Type: mysql.TypeDatetime, Fsp: 6}},
```

Comment:
Why not add the test data which is in #9039 ?

Reply:
just want to be consistent with the test sample above :)